src/renderer/sw_engine/tvgSwStroke.cpp

```diff
@@ -860,7 +861,7 @@ bool strokeParseOutline(SwStroke* stroke, const SwOutline& outline)
         if (type == SW_CURVE_TYPE_CUBIC) return false;
         ++types;
 
-        auto closed =  outline.closed.data ? outline.closed.data[i]: false;
+        auto closed =  outline.closed.data ? outline.closed.data[i - 1]: false;
```

Comment:
Thanks, may I ask why you changed the logic? I think this should remain unchanged to preserve the original logic.

Reply:
Because I added `++i` at the very beginning of the loop, so the first value of i is 1. This is because we no longer increment `i` in the for statement but inside the loop body. Since there is a continue statement, I didn’t want to repeat `++i` in two places. Instead, I ensured it always happens by placing it at the very beginning of the loop, but as a result, I had to adjust this line.
do you think it will be more clear to keep as it was and repeat `++i` in two places?